src/history/hs.c

```diff
@@ -1088,6 +1094,14 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
             WT_STAT_DATA_INCR(session, cursor_skip_hs_cur_position);
 
             WT_ERR(__wt_compare(session, NULL, &cursor->key, srch_key, &cmp));
+            WT_ERR(cursor->get_key(cursor, &ds_btree_id, &ds_key, &ds_start_ts, &counter));
+
+            /* Stop before crossing over to the next btree */
+            if (ds_btree_id != btree_id) {
```

Comment:
Wouldn't you expect `cmp <= 0` in this case? I don't get why we need to make this a separate check.

Also, the point of putting this `prev` in a loop is that you may be more than 1 record away from your desired key range due to racing with inserts. I think that if you race with an insert from an adjacent btree, you'll potentially just break out of here.

For example, say we're positioning for: `btree_id=2, key=foo, ts=MAX`.
```
btree_id key ts
2        aaa  2
2        bbb  2
2        bbb  3
2        bbb  4
2        bbb  5
3        bbb  2 <- cursor positioned here after search_near
```
When when we're about to do `prev`, we can get another record in between like this:
```
btree_id key ts
2        aaa  2
2        bbb  2
2        bbb  3
2        bbb  4
2        bbb  5
3        aaa  2 <- insert raced here
3        bbb  2 <- cursor pointed here
```

Reply:
Oh I see so you're expecting to land on the btree below ours, hence why we don't do the btree comparison. But if you never landed on your btree then when is cmp <= 0. This loop would just run of the top of the history store right?